### PR TITLE
Rework the New Architecture Section

### PR DESCRIPTION
**Pull request by lunaleaps**
## [See updated doc](https://deploy-preview-3998--react-native.netlify.app/docs/next/the-new-architecture/landing-page)

And we moved the rest of the New Architecture docs to the working group. See https://github.com/reactwg/react-native-new-architecture/pull/156

### CODE REVIEW

File: docs/the-new-architecture/landing-page.md
Comment (mrousavy) [inline docs/the-new-architecture/landing-page.md:47]:
```suggestion

### Compile-time code generation

CodeGen, an important part of the new architecture, generates types for the bindings between JavaScript and native TurboModules.

Using this generated type information, parameters and return values from native TurboModules receive proper nullability annotations as well as type declarations, resulting in potentially safer code.

Additionally, over-the-air updates can be automated easier, since CodeGen can be used to determine whether a native module's call-signature has been changed and a native update is required.

```

Something along the lines of this might also be good mentioning? CodeGen is pretty dope once it matures imo

File: docs/the-new-architecture/landing-page.md
Comment (mrousavy) [inline docs/the-new-architecture/landing-page.md:45]:
```suggestion

Similar to VisionCamera's frame buffers, JSI can be used to expose other complex instance-based types such as databases, images, audio samples, and more.

```

Maybe also worth mentioning that HostObjects are a really cool thing that could become even more popular in the future, e.g. react-native-mmkv is a single JS object that is actually a C++ instance. This is huge, because previously all native modules are just static methods. Now you can create instances, e.g. for databases, images, audio samples, or the frame buffers as in VisionCamera. Maybe even make File I/O native modules safer as a file can be opened, etc etc

**Review by mrousavy — COMMENTED** [review]
looks dope luna - left a few comments / thoughts :)

File: docs/the-new-architecture/landing-page.md
Comment (lunaleaps) [reply docs/the-new-architecture/landing-page.md:44]: Love it, I agree the twitter handle might feel a little-out-of-place but great suggestions. Let me incorporate

File: docs/the-new-architecture/landing-page.md
Comment (lunaleaps) [reply docs/the-new-architecture/landing-page.md:47]: I agree codegen is pretty neat, do you feel like it's more interesting than the other points mentioned? I want to keep this section relatively short and we can always go into more depth on the working group.

Comment (mrousavy) [reply docs/the-new-architecture/landing-page.md:48]: I don't think it is _more interesting_ than the other points mentioned, but I do think that this allows for many possibilities such as OTA, and in general a much safer native interop. This has been an issue a lot in many libraries, where e.g. the JS bundle is out of sync with native code.

But I agree that the other points are more interesting and this _could_ be left out, your call :)

File: docs/the-new-architecture/landing-page.md
Comment (rickhanlonii) [inline docs/the-new-architecture/landing-page.md:16]: Can we include a code example with the before/after? I think that will make the value "click" better.

File: docs/the-new-architecture/landing-page.md
Comment (rickhanlonii) [inline docs/the-new-architecture/landing-page.md:20]: Can we link to the bugs as examples? I'm sure these have been reported in the issue tracker.

File: docs/the-new-architecture/landing-page.md
Comment (rickhanlonii) [inline docs/the-new-architecture/landing-page.md:22]: Can we include code for the example? It's not clear to me which feature this demo is using. It makes it sound as if this is an improvement/breakage that you will get out of the box (untrue) vs the reality that you opt-in with a feature.

File: docs/the-new-architecture/landing-page.md
Comment (rickhanlonii) [inline docs/the-new-architecture/landing-page.md:42]: I feel like we're going out of our way to say "bridge" here but I think we should call it out because people are well aware of the bridge and we explained this as "removing the async bridge" when we [announced this project in 2018](https://reactnative.dev/blog/2018/06/14/state-of-react-native-2018).

File: docs/the-new-architecture/landing-page.md
Comment (rickhanlonii) [inline docs/the-new-architecture/landing-page.md:51]: Link?

File: docs/the-new-architecture/landing-page.md
Comment (mrousavy) [reply docs/the-new-architecture/landing-page.md:42]: I agree with you here but it would technically be wrong terminology, no? 

JSI is still a _bridge_ between JS and native, just like the old JSON message queue was a _bridge_.

I definitely think people will understand it though, so maybe just put it in quotes or add a hyperlink to it that people can click to read about the bridge?

Comment (cortinico) [reply docs/the-new-architecture/landing-page.md:42]: +1 for calling out bridge here as it's part of common knowledge at this point

File: docs/the-new-architecture/landing-page.md
Comment (cortinico) [inline docs/the-new-architecture/landing-page.md:72]: This link lands to a Table of Content. I'd say we would need more of a "How to enable the New Architecture" kind of page on the WG end

File: docs/the-new-architecture/landing-page.md
Comment (cortinico) [inline docs/the-new-architecture/landing-page.md:71]: Commenting here but valid to the WG: should we make the New Architecture WG open. Currently users have to go over a Google Fork (linked in the readme) to be included

**Review by cortinico — COMMENTED** [review]
Great stuff!

File: docs/the-new-architecture/landing-page.md
Comment (lunaleaps) [reply docs/the-new-architecture/landing-page.md:72]: I need to actually move those docs over and will update the link to the actual instructions

File: docs/the-new-architecture/landing-page.md
Comment (lunaleaps) [reply docs/the-new-architecture/landing-page.md:71]: I think that's a good idea, maybe that's something we can remove with this doc landing

File: docs/the-new-architecture/landing-page.md
Comment (lunaleaps) [reply docs/the-new-architecture/landing-page.md:22]: I think that is the case though here, where using the new renderer is throwing away intermediate work. Let me try and emphasize that more here

File: docs/the-new-architecture/landing-page.md
Comment (cortinico) [inline docs/the-new-architecture/landing-page.md:223]: The only nit I have here:

I would create a:
```
### Enabling the New Architecture
```

heading before this paragraph as I can imagine is one of the thing folks will search for. I'm wondering if we even need the one-liner on how to do it in Android/iOS here or if we just want to let the user jump to the WG directly

**Review by cortinico — APPROVED** [review]
LGTM 👍 Accepting to unblock you @lunaleaps this is amazing work!

File: docs/the-new-architecture/landing-page.md
Comment (Simek) [inline docs/the-new-architecture/landing-page.md:33]:
```suggestion
```jsx
```
Syntax highlight nit.

File: docs/the-new-architecture/landing-page.md
Comment (Simek) [inline docs/the-new-architecture/landing-page.md:59]:
```suggestion
```jsx
```
Syntax highlight nit.

File: docs/the-new-architecture/landing-page.md
Comment (Simek) [inline docs/the-new-architecture/landing-page.md:131]:
```suggestion
```jsx
```

Syntax highlight nit.

File: docs/the-new-architecture/landing-page.md
Comment (lunaleaps) [reply docs/the-new-architecture/landing-page.md:225]: Let's keep the instructions in the WG for now, so we don't have to update multiple places

File: website/static/_redirects
Comment (lunaleaps) [inline website/static/_redirects:39]: What's really odd is that these don't work, even when when I explicitly write them out 🤔
See related PR: https://github.com/facebook/react-native-website/pull/4029

File: docs/the-new-architecture/landing-page.md
Comment (lunaleaps) [inline docs/the-new-architecture/landing-page.md:208]: @NickGerleman Would this be a good link?

File: website/static/_redirects
Comment (Simek) [reply website/static/_redirects:39]: Hmm, It's only a guess, but maybe you need also to remove the pages from versioned sidebars and versioned docs, so the Docusuarus do not try to overwrite or modify the redirects in any way? 

* https://github.com/facebook/react-native-website/tree/main/website/versioned_sidebars
* https://github.com/facebook/react-native-website/tree/main/website/versioned_docs

Looks like this PR only updates the "next" sidebar, and do not touch pages. Spotted also that we lack redirect for 0.70 docs (not all avaible version are deployed in previews, to optimize the build time.

File: docs/the-new-architecture/landing-page.md
Comment (lunaleaps) [reply docs/the-new-architecture/landing-page.md:208]: Discussed offline, this link is good for now

File: website/static/_redirects
Comment (lunaleaps) [reply website/static/_redirects:39]: Hmm let me try that.

Ah good catch. I'll update the other non-next sidebars as well and I guess the relevant intro page

File: website/static/_redirects
Comment (lunaleaps) [inline website/static/_redirects:45]:
```suggestion
/docs/:version/the-new-architecture/cxx-custom-types https://github.com/reactwg/react-native-new-architecture#guides
```

File: website/static/_redirects
Comment (lunaleaps) [reply website/static/_redirects]: I don't think it's the matter of deleting the docs ... also I think in the end I don't want to delete right now because we do link to them in some blog posts. Can probably do a clean-up job one day.

I also thought it might be this build step around updating _redirects https://github.com/facebook/react-native-website/pull/3680 but inspecting the _redirects file generated from `yarn build`, I don't see how that affects the version. 

I also tried deleting the `$LATEST_VERSION$` redirect. Seriously confused 😵

I think for now, I'll go ahead and land. 0.72 docs will somehow always point to the old New Architecture docs 🤷‍♀️

**Pull request merged**
